bx_py_utils/test_utils/snapshot.py

```diff
@@ -0,0 +1,24 @@
+import json
+import pathlib
+import re
+from typing import Union
+
+
+def assert_snapshot(root_dir: Union[pathlib.Path, str], snapshot_name: str, got: Union[dict, list]):
+    assert re.match(r'^[-_.a-zA-Z0-9]+$', snapshot_name), f'Invalid snapshot name {snapshot_name}'
+    assert isinstance(got, (dict, list))
+
+    snapshot_file = pathlib.Path(root_dir) / f'{snapshot_name}.json'
+    try:
+        with snapshot_file.open('r') as snapshot_handle:
+            expected = json.load(snapshot_handle)
+    except (ValueError, IOError, FileNotFoundError, OSError):
+        with snapshot_file.open('w') as snapshot_handle:
+            json.dump(got, snapshot_handle)
```

Comment:
What's about to add: `ensure_ascii=False, sort_keys=True, indent=4` ?

Same below or better merge into a function?

Reply:
Very good points, totally forgot! Added all.